### ISSUE TITLE
docs: Encourage `pipx` more strongly in installation guide

### ISSUE THREAD

**Issue by aaronsteers** [meltano/meltano#6882]
I love the new installation guide from @sbalnojan . However, one thing we lost in the transition to the new format was the prioritization of `pipx` as the preferred installation method - and the instructions to leverage virtual environments if `pip` is used directly.

1. I would suggest we change the order of the tabs from [1: `pip`, 2: `docker`, 3: `pipx`] to instead use: [1: `pipx`, 2: `pip`, 3: `docker`]. People who want to do a 'quick test' and don't want to install pipx can still just jump over to the `pip` tab, and they can ignore the warnings/recommendations, but then if they do run into version conflicts, they remember ignoring the warnings and they know to come back and run it in one of the recommended methods.

2. The `pipx` section links to the `pipx` homepage, but it could also be helpful to link to the 'how to install pipx' content directly here: https://docs.meltano.com/guide/installation-guide#install-pipx

3. The `pip` section also currently suggests a simple installation, which on its own is very likely to lead to version conflicts. I think we should definitely recommend virtual environments on the `pip` tab, ideally with a warning about version conflicts, suggesting pip+virtualenv or pipx as the recommended path to avoid conflicts with other installed tools and libraries. To avoid blowing up the page with `virtualenv` explanation and instructions there on the `pip` tab, I think we could just add something like this with links to content off the page:

> Warning: We strongly recommend you use `pipx` for installation. If you choose to use `pip` directly, we recommend creating a dedicated virtual environment to avoid conflicts with other installed software. [Learn more here.](https://docs.meltano.com/concepts/python_virtual_environments#what-are-python-virtual-environments)

cc @tayloramurphy

**Comment by tayloramurphy** [comment]
@aaronsteers I think that makes sense. Sven was prioritizing speed of install generally, but if that comes at the expense of some downstream problems it may be worth taking a moment to recommend pipx. @sbalnojan thoughts?

**Comment by aaronsteers** [comment]
Yeah, @tayloramurphy - totally understood. I was mindful not to suggest slowing down the innovation/momentum in that PR because of how much I loved the new tabbed experience - and because I think the simplication in itself was a huge win.

But yeah, also, I would like to make sure we don't leading folks down a path that will hit version conflicts or other spooky issues. 👻

**Comment by sbalnojan** [comment]
@aaronsteers & @tayloramurphy, so I don't disagree, and I think we can make all of it work if we're deliberate about it.

I would however like to put a few questions out there because I'm not sure on the right direction here.
- What's the level of Python we assume the Meltano newbie has? I'm talking about the (open source) target audience, the one that is able to utilise Meltano.
- Is Meltano in some way more susceptible to dependency troubles than other Python-based tools? 
- What's our assumption on the audiences pipx familarity (50% have heard of it, 25% have it installed?)?

**Comment by aaronsteers** [comment]
> @aaronsteers & @tayloramurphy, I don't disagree, and I think we can make all of it work if we're deliberate about it.
> 
> I would however like to put a few questions out there because I'm not sure on the right direction here.
> - What's the level of Python we assume the Meltano newbie has? I'm talking about the (open source) target audience, the one that is able to utilise Meltano.

I assume zero Python experience for a Meltano newbie. Maybe 40-60% will happen to have prior python experience when they find us, but I don't see python as a requirement at all - for the purposes of the install process, it's just a delivery mechanism. Hopefully Meltano users never have to write python unless they want to build a tap that doesn't exist yet - and even then, they can learn python as they go with the help up the cookie cutter templates.

> - Is Meltano in some way more susceptible to dependency troubles than other Python-based tools? 

Yes, actually we are. As a Python based tool that runs other python tools, Meltano often gets installed on machines that already have airflow or dbt installed, for instance. Those installations will fail and users will ping is in slack or create PRs trying to reconcile version dependencies between Meltano and tool X.
 
> - What's our assumption on the audiences pipx familarity (50% have heard of it, 25% have it installed?)?

Maybe 10-25% will have heard of it? But fewer than that know how to setup virtual environments manually. And after installing manually with virtual environments, they still need to remember to reactivate the environment each time. This is the reason pipx exists: you get the simplicity of pip with the safety of virtual environments. The point is to give the simplest _safe_ install method. The user can take short cuts, of course, but if they run into trouble, they'll just retry with the correct/safe method, rather than think Meltano is broken.

**Comment by sbalnojan** [comment]
@aaronsteers Ok there's the rub. I was assuming a very different profile. Then we do need to change something, not sure yet how. 

My gut feeling would be to then indeed push the pipx version, simply because teaching someone to handle venvs is hard, and would break the flow here. Better to use pipx then and simply "get it to work", while also putting a clear disclaimer on top ("if you know how to handle venvs, just go ahead and do X; If not, here's the easy way: ..."). 

That would still keep the discussion of venvs out of the quick install as well as most of the indepth install content.

**Comment by aaronsteers** [comment]
@sbalnojan - Yes, makes sense to me 👍

**Comment by cwegener** [comment]
I just noticed this issue and didn't realise until now that there was a documentation update which lost this preference of `pipx` over any other installation method. Just wanted to highlight the first case of a `pip install` in the wild ... https://meltano.slack.com/archives/CMN8HELB0/p1672953079682619

Whilst looking through the `pip install` flow for meltano installation myself, I also noticed that `virtualenv` is actually not an explicit dependency in the `pyproject.toml` file, which is actually not an issue with the `pipx` method. Because `pipx` takes care of making sure that `virtualenv` is present. But that's not the case for the `pip install` method. Users can easily have just `pip` installed but not `virtualenv`
